syntax = "proto3";

// A compact summary of a single VIDEO track's key features.
message VideoSummary {
  string codec = 1;                   // e.g., "HEVC", "AVC"
  int32 bit_depth = 2;                // e.g., 10, 8
  int32 width = 3;                    // e.g., 3840
  int32 height = 4;                   // e.g., 2160
  string hdr = 5;                     // e.g., "DV, HDR", "SDR"
  bool is_3d = 6;                     // Is it a 3D video stream (e.g., H-SBS)?
  bool has_hardcoded_subtitles = 7;   // Does this video track contain hardcoded subtitles?
  string source = 8;                  // e.g., "Warner Bros. USA UHD Blu-ray (2025)"
}

// A compact summary of a single AUDIO track's key features.
message AudioSummary {
  string language = 1;                // e.g., "English", "Japanese"
  string format_tag = 2;              // e.g., "Atmos", "TrueHD", "DTS-HD MA", "FLAC"
  string channels_tag = 3;            // e.g., "7.1", "5.1", "2.0"
  bool is_commentary = 4;             // Is this an audio commentary track?
  bool is_descriptive = 5;            // Is this a descriptive audio track?
  string source = 6;                  // e.g., "Warner Bros. USA UHD Blu-ray (2025)"
}

// A compact summary of a single SUBTITLE track's key features.
message SubtitleSummary {
  string language = 1;                // e.g., "English", "French"
  string format = 2;                  // e.g., "PGS", "SRT"
  bool is_sdh = 3;                    // Subtitles for the Deaf and Hard of Hearing?
  string source = 4;                  // e.g., "iTunes WEB-DL"
}

// The main "Hot" data object to be stored in KeyDB.
// Contains all the essential metadata parsed from the media file and its name.
message MediaInfoSummary {
  // --- Core Media Identifiers ---
  string title = 1;                   // The title of the movie or episode.
  string imdb_id = 2;                 // e.g., "tt31193180"
  string tmdb_id = 3;                 // e.g., "movie/1233413"
  string unique_id = 4;               // The 128-bit unique ID from the media container (e.g., MKV). Optional.
  string torrent_hash = 5;            // The SHA256 hash of the torrent.
  string edition = 6;                 // e.g., "Director's Cut", "Extended Edition", "Uncut"
  string group = 7;                   // The release group, e.g., "CiNEPHiLES"
  string quality = 8;                 // e.g., "BluRay REMUX", "WEB-DL"
  string container = 9;               // The file container, e.g., "mkv", "mp4"
  int64 size = 10;                    // File size in bytes. Use int64 for large files.
  string site = 11;                   // The site the media originated from.
  string network = 12;                // The TV network or streaming service.

  // --- Boolean Flags ---
  bool extended = 13;                 // Is this an extended version?
  bool uncensored = 14;               // Is the content uncensored?
  bool upscaled = 15;                 // Is the video an upscale?

  // --- TV Show Specific Fields (Hybrid Approach) ---
  int32 season_number = 16;           // The parsed season number, e.g., 1. Use 0 for specials.
  int32 episode_number = 17;          // The parsed episode number, e.g., 1.
  string episode_string = 18;         // The original display string, e.g., "S01E01", "S01E01-E02"

  // --- Torrent Specific Field ---
  int32 torrent_file_index = 19;      // The index of this file within the torrent (0, 1, 2...).

  // --- Track Summaries (These are dynamic lists) ---
  repeated VideoSummary video_tracks = 20;
  repeated AudioSummary audio_tracks = 21;
  repeated SubtitleSummary subtitle_tracks = 22;
}